code/game/machinery/atmoalter/clamp.dm

```diff
@@ -142,9 +142,12 @@
 
 	if (istype(A, /obj/machinery/atmospherics/pipe/simple))
 		to_chat(user, "<span class='notice'>You begin to attach \the [src] to \the [A]...</span>")
-		if (do_after(user, 30, src))
+		var/C = locate(/obj/machinery/clamp) in get_turf(A)
+		if (do_after(user, 30, src) && !C)
 			if(!user.unEquip(src))
 				return
 			to_chat(user, "<span class='notice'>You have attached \the [src] to \the [A].</span>")
 			new/obj/machinery/clamp(A.loc, A)
-			qdel(src)
\ No newline at end of file
+			qdel(src)
+		if(C)
```

Comment:
this should go above the do_after() and have a return. At present it finds a clamp, lets you attach another, then tells you there was already one there, without removing the new one

Reply:
146 has a check to ensure C does not exist, this `if(C)` is after to ensure the do_after has a chance to cycle before saying it fails. As in, the person tries to attach it, and then fails.

Reply:
oh, I'm just blind